pkg/catalog/catalog.go

```diff
@@ -2765,6 +2765,29 @@ func (c *Catalog) PrepareGCUncommitted(ctx context.Context, repositoryID string,
 	}, nil
 }
 
+func (c *Catalog) CloneEntry(ctx context.Context, srcRepository, srcRef, srcPath, destRepository, destBranch, destPath string, opts ...graveler.SetOptionsFunc) (*DBEntry, error) {
```

Comment:
Why not change `CopyEntry` behavior and create `CloneEntry`?

Reply:
@itaiad200 do you have a suggestion when to verify that the copy is a safe clone on lakeFS?